bin/main.rb

```diff
@@ -1,87 +1,34 @@
-# !/usr/bin/env ruby
-puts 'Welcome To Tic Tac Toe Game'
+#!/usr/bin/env ruby
+require_relative '../lib/tic_tac_toe.rb'
+require_relative '../lib/validatable.rb'
 
-def draw_board(tile)
-  "\n-----------------------------------------\n\n" \
-  "#{tile[0]} | #{tile[1]} | #{tile[2]}\n" \
-  "#{tile[3]} | #{tile[4]} | #{tile[5]}\n" \
-  "#{tile[6]} | #{tile[7]} | #{tile[8]}\n" \
-  "\n-----------------------------------------\n\n"
-end
-
-def won?(symbole, tile)
-  winning_combination = [
-    [0, 1, 2], [3, 4, 5], [6, 7, 8],
-    [0, 3, 6], [1, 4, 7], [2, 5, 8],
-    [0, 4, 8], [2, 4, 6]
-  ]
-  won = false
-  winning_combination.each do |row|
-    won = true if tile[row[0]] == symbole && tile[row[1]] == symbole && tile[row[2]] == symbole
-  end
-
-  won
-end
-
-player = {
-  '1': nil,
-  '2': nil
-}
+puts <<~HEARDOC
+  ***************************************
+  *                                     *
+  *           TIC TAC TOE               *
+  *                                     *
+  ***************************************
+HEARDOC
 
-symboleee = {
-  '1': nil,
-  '2': nil
-}
+game = TicTacToe.new
 
-symboleee[':1'] = 'x'
-symboleee[':2'] = 'O'
-puts 'Player 1 Name'
-
-player[':1'] = gets.chomp
-puts 'Player 2 Name'
-player[':2'] = gets.chomp
+game.add_players
 
 continue = true
-turn = 1
-# rubocop: disable Metrics/BlockNesting
 while continue
-  round = 0
-  tile = [1, 2, 3, 4, 5, 6, 7, 8, 9]
-  current_symbole = ''
-  while round < 9
-    puts 'Round ' + round.to_s
-    puts draw_board(tile)
-    turn = turn == 1 ? 2 : 1
-    puts player[":#{turn}"] + '\'s Turn'
-
-    choice = false
-    until choice
-      choice = gets.chomp
-      valid = choice.match(/[1-9]/)
-      choice = valid ? choice.to_i : false
-      puts 'invalid input \n valid input is a number between 1 and 9' unless choice
-    end
-    current_symbole = turn == 1 ? symboleee[':1'] : symboleee[':2']
-    tile[choice - 1] = current_symbole
-    round += 1
-
-    next unless round > 4 && won?(current_symbole, tile)
-
-    winner = player[":#{turn}"]
-    puts '************************************************************'
-    puts '* Congradulation *'
-    puts " #{winner}  has won"
-    puts '************************************************************'
+  game.play
+  puts 'Game Over \n Do you want to play another round (y/n)'
+
+  loop do
+    begin
+    play = gets.chomp
+    Validatable.exit_choice_valid?(play)
+    continue = play.downcase != 'n'
```

Comment:
Your loop's indentation is not correct. Please make it right so the code is more readable.

Reply:
**Fixed**